src/lib/core/datetime.c

```diff
@@ -88,9 +89,12 @@ tm_to_datetime(struct tnt_tm *tm, struct datetime *date)
 			dt = ((wday - 4) % 7) + DT_EPOCH_1970_OFFSET;
 		}
 	} else {
-		assert(mday >= 0 && mday < 32);
+		if (mday == 0)
+			mday = 1;
+		assert(mday >= 1 && mday <= 31);
 		assert(mon >= 0 && mon <= 11);
-		dt = dt_from_ymd(year + 1900, mon + 1, mday);
+		if (dt_from_ymd_checked(year + 1900, mon + 1, mday, &dt) == false)
```

Comment:
You should probably squash this commit (164353c1) to 3c585e2.

Reply:
I'd rather not - I've kept it separate because in the single commit we add huge years support both for default `datetime_parse_full` and for format-based `tnt_strptime`.